.circleci/config.yml

```diff
@@ -1,56 +1,378 @@
       - image: circleci/postgres:9.6.11-postgis
         environment:
           POSTGRES_USER: "test"
           POSTGRES_PASSWORD: "test"
           POSTGRES_DB: "test"
-      - image: circleci/mongo:3.4.18
-      # - image: mcr.microsoft.com/mssql/server:2017-GA-ubuntu
-      #   environment:
-      #     SA_PASSWORD: "Admin12345"
-      #     ACCEPT_EULA: "Y"
-      # - image: cockroachdb/cockroach-unstable:latest
-      #   command: start --insecure
+  sqlite:
+    docker:
+      - image: circleci/node:10.15.0 # sqlite doesn't require another image
+  sqljs:
+    docker:
+      - image: circleci/node:10.15.0 # sqljs doesn't require another image
 
+commands:
+  prepare-for-tests:
+    parameters:
+      node_folder:
+        type: enum
+        enum:
+          - node8
+          - node10
+          - node12
     steps:
-      - checkout
+      - attach_workspace:
+          at: ~/typeorm/
       - run: sudo npm install -g npm@latest
       - run: cp ormconfig.circleci.json ormconfig.json
 
       # Download and cache dependencies
       - restore_cache:
           keys:
-          - v1-dependencies-{{ checksum "package.json" }}
-          # fallback to using the latest cache if no exact match is found
-          - v1-dependencies-
+            - << parameters.node_folder >>-v1-dependencies-{{ checksum "package.json" }}
+            # fallback to using the latest cache if no exact match is found
+            - << parameters.node_folder >>-v1-dependencies-
 
       - run: npm install
 
       - save_cache:
           paths:
             - node_modules
-          key: v1-dependencies-{{ checksum "package.json" }}
-
-      # - run: docker-compose up -d
+          key: << parameters.node_folder >>-v1-dependencies-{{ checksum "package.json" }}
 
       # run tests
       - run: npm run lint
+
+  run-db-tests:
+    parameters:
+      database:
+        type: enum
+        enum:
+          - mariadb
+          - mongodb
+          - mssql
+          - mysql
+          - oracle
+          - postgres
+          - sqlite
+          - sqljs
+    steps:
+      - attach_workspace:
+          at: ~/typeorm/
+      - run: cp ormconfigs/ormconfig.<< parameters.database >>.json ormconfig.json
       - run: npm test
+
+jobs:
+  checkout_code:
+    docker:
+      - image: circleci/node:10.15.0
+    working_directory: ~/typeorm/checkout/
+    steps:
+      - checkout
+      - run:
+          name: Copy source for node versions
+          command: |
+            mkdir ~/typeorm/node8/
+            mkdir ~/typeorm/node10/
+            mkdir ~/typeorm/node12/
+
+            cp -R ~/typeorm/checkout/. ~/typeorm/node8/
+            cp -R ~/typeorm/checkout/. ~/typeorm/node10/
+            cp -R ~/typeorm/checkout/. ~/typeorm/node12/
+      - persist_to_workspace:
+          root: ~/typeorm/
+          paths:
+            - ./checkout/
+            - ./node8/
+            - ./node10/
+            - ./node12
+
+  test-node8: &test-template
```

Comment:
This step's name is a misnomer that I need to fix. It actually leverages the `prepare-for-tests` command which runs `npm install`. In this case, it is for Node 8. The `&test-template` is a YAML feature which allows this YAML object to be referenced later on and copied, so that the Node 10 & 12 respective steps only need to specify the values that are different. Right now, that actually means everything, since each property has node version specific values, but it could be used to only specify some values once.